plugins/grid-bookmark/src/GridBookmarkWidget/model.ts

```diff
@@ -150,12 +172,43 @@ export default function f(_pluginManager: PluginManager) {
       ) {
         bookmark.correspondingObj.setLabel(label)
       },
+      updateBookmarkHighlight(
+        bookmark: IExtendedLabeledRegionModel,
+        color: string,
+      ) {
+        bookmark.correspondingObj.setHighlight(color)
+      },
+      updateBulkBookmarkHighlights(color: string) {
+        if (self.selectedBookmarks.length !== 0) {
+          self.selectedBookmarks.forEach(bookmark =>
+            this.updateBookmarkHighlight(bookmark, color),
+          )
+        }
+      },
       setSelectedBookmarks(bookmarks: IExtendedLabeledRegionModel[]) {
         self.selectedBookmarks = bookmarks
       },
       setBookmarkedRegions(regions: IMSTArray<typeof LabeledRegionModel>) {
         self.bookmarks = cast(regions)
       },
+      setHighlightToggle(toggle: boolean) {
+        const { views } = getSession(self)
+        const lgvs = views.filter(v =>
```

Comment:
consolidating into a filtered array may be unneeded. could use views.forEach(view => v.showBookmarkHighlights?.(toggle))